server/src/main/java/io/spine/server/entity/TransactionalEventPlayer.java

```diff
@@ -0,0 +1,63 @@
+/*
+ * Copyright 2018, TeamDev. All rights reserved.
+ *
+ * Redistribution and use in source and/or binary forms, with or without
+ * modification, must retain the above copyright notice and the following
+ * disclaimer.
+ *
+ * THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS
+ * "AS IS" AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT
+ * LIMITED TO, THE IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR
+ * A PARTICULAR PURPOSE ARE DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT
+ * OWNER OR CONTRIBUTORS BE LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL,
+ * SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT
+ * LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES; LOSS OF USE,
+ * DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND ON ANY
+ * THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT
+ * (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE
+ * OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+ */
+
+package io.spine.server.entity;
+
+import io.spine.core.EventEnvelope;
+import io.spine.server.event.EventStream;
+
+import static com.google.common.base.Preconditions.checkNotNull;
+
+/**
+ * An {@link EventPlayer} which plays events upon the given {@link Transaction}.
+ *
+ * @author Dmytro Dashenkov
+ */
+final class TransactionalEventPlayer implements EventPlayer {
+
+    private final Transaction<?, ?, ?, ?> transaction;
+
+    /**
+     * Creates a new instance of {@code TransactionalEventPlayer}.
+     *
+     * @param transaction the transaction
+     */
+    TransactionalEventPlayer(Transaction<?, ?, ?, ?> transaction) {
+        this.transaction = checkNotNull(transaction);
+    }
+
+    /**
+     * Plays the given events upon the underlying entity transaction.
+     *
+     * @param events {@inheritDoc}
+     */
+    @Override
+    public void play(EventStream events) {
+        events.events()
+              .stream()
+              .map(EventEnvelope::of)
+              .forEach(transaction::apply);
+    }
+
+    @Override
+    public String toString() {
+        return "EventPlayer.forTransactionOf(...)";
```

Comment:
This is weird. 

Let's print something that:
* points to the fact that this is an `TransactionalEventPlayer`;
* mentions the `transaction` — I'd override the `Transaction.toString()` as it's its responsibility to print itself.